### PR TITLE
RWA-1745: Refactor functional tests for when initiation CRON job is switched on on AAT.

### PR DESCRIPTION
**Pull request by rasikasri**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-1745

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/testUtils/java/uk/gov/hmcts/reform/wataskmanagementapi/SpringBootFunctionalBaseTest.java [hunk 1]
```diff
@@ -183,78 +206,128 @@ public String createTask(CreateTaskMessage createTaskMessage) {
...
+                                Map<String, String> additionalProperties) {
+        initiateTask(testVariables, headers, additionalProperties, defaultInitiationAssert(testVariables));
+    }
+
+    private void initiateTask(TestVariables testVariables,
+                              Headers headers,
+                              Map<String, String> additionalProperties,
+                              Consumer<Response> assertConsumer) {
+        log.info("Task initiate with cron job {}", initiationJobRunning);
+        if (initiationJobRunning) {
```
Comment (serdaroguzhanakin) [inline src/testUtils/java/uk/gov/hmcts/reform/wataskmanagementapi/SpringBootFunctionalBaseTest.java:251]: initiation process has to wait for task initiation job in AAT, I guess we need to wait for task initiation and then check it's in CFT task DB. If I'm wrong please let me know. This is my suggestion for a delay
this waits 2 seconds before DB check

with()
                .pollInterval(2, SECONDS)
                .and()
                .with()
                .pollDelay(2, SECONDS).
                await()
                .atMost(60, SECONDS)
                .until(
                    () -> {
                        Response response = restApiActions.get(
                            TASK_GET_ENDPOINT,
                            testVariables.getTaskId(),
                            headers
                        );

                        return HttpStatus.OK.value() == response.getStatusCode();
                    }
                );

File: src/testUtils/java/uk/gov/hmcts/reform/wataskmanagementapi/utils/Common.java [hunk 2]
```diff
@@ -205,84 +217,105 @@ public TestVariables setupTaskWithoutCcdCaseAndRetrieveIdsWithCustomVariable(
...
             .and()
-            .iRetrieveATaskWithProcessVariableFilter("caseId", caseId, 1);
+            .iRetrieveATasksWithProcessVariableFilter("caseId", caseId, taskType);
 
         if (response.size() > 1) {
             fail("Search was not an exact match and returned more than one task used: " + caseId);
         }
-
-        return new TestVariables(caseId, response.get(0).getId(), response.get(0).getProcessInstanceId());
+        // return taskId
```
Comment (serdaroguzhanakin) [inline src/testUtils/java/uk/gov/hmcts/reform/wataskmanagementapi/utils/Common.java:250]: missed remove

Comment (rasikasri) [reply src/testUtils/java/uk/gov/hmcts/reform/wataskmanagementapi/utils/Common.java:250]: Yes

**Review by serdaroguzhanakin — DISMISSED** [review]
LGTM

**Review by serdaroguzhanakin — DISMISSED** [review]
LGTM

**Review by serdaroguzhanakin — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

**Pull request merged**